ACMILabs/interactive-label
# #100 Implemented unsuccessful tap UI

Resolves #100 

When a tap fails to be created in XOS, an error dialogue now displays.

### Acceptance Criteria
- [x] Failure dialogue appears on failed tap
- [x] Dialogue matches design files

### Relevant design files
* https://trello.com/c/Y2He78ze/142-fix-playlist-label-interactive-label-and-spotlights-to-return-an-error-message-instead-of-collected-when-tapping-with-a-broken-l

### Testing instructions
1. Modify your config.env
   - XOS_API_ENDPOINT=https://xos.acmi.net.au/api/
   - XOS_TAPS_ENDPOINT=http://your-ip-address:5000/api/taps/
   - PLAYLIST_ID=44
2. Run interactive-label i.e. `cd development && docker-compose up`
3. Run your development lens reader on [e__taps-reader-pi-4](https://dashboard.balena-cloud.com/devices/81361d59c71358c33710191666890fe8/envvars) and modify it's environment variable to point to your spotlights
   - TARGET_API_ENDPOINT=http://your-ip-address:8081/api/
4. Run your favourite flask tap server! (or modify the one I used below with your own valid lens uids).
5. Open up your browser at localhost:8081, set the dimensions of the browser to 1920x1080 in the web console
5. Tap with a valid lens, notice the normal 'collected' UI response occurs
6. Tap with an invalid lens, notice the new error dialogue

### Notes
1. I edited the flask server to emulate invalid lenses. Feel free to do so as well (replacing the 'valid-lens-uid'), or of course use your own method.
```
# server.py
from flask import Flask, Response, request
app = Flask(__name__)

@app.route('/')
def home():
    return taps()

@app.route('/api/taps/', methods=['POST'])
def taps():
    response_json = request.get_json()
    print(response_json)
    lens_uid = response_json['lens']['uid']
    status_code = 201 if lens_uid in ['<valid-lens-uid>'] else 400
    return Response('{"ok": "true"}', status=status_code, mimetype='application/json')
```

### DoD
